test/Sentry.AspNetCore.Tests/ScopeExtensionsTests.cs

```diff
@@ -0,0 +1,478 @@
+
+        Assert.DoesNotContain("REMOTE_ADDR", _sut.Request.Env.Keys);
+    }
+
+    [Fact]
+    public void Populate_RemoteIp_SendDefaultPiiTrue_SetToEnv()
+    {
+        const string expected = "::1";
+        var connection = Substitute.For<ConnectionInfo>();
+        connection.RemoteIpAddress.Returns(IPAddress.IPv6Loopback);
+        _httpContext.Connection.Returns(connection);
+
+        SentryAspNetCoreOptions.SendDefaultPii = true;
+        _sut.Populate(_httpContext, SentryAspNetCoreOptions);
+
+        Assert.Equal(expected, _sut.Request.Env["REMOTE_ADDR"]);
+    }
+
+    [Fact]
+    public void Populate_LocalPort_SetToEnv()
+    {
+        const int expected = 1337;
+        var connection = Substitute.For<ConnectionInfo>();
+        connection.LocalPort.Returns(expected);
+        _httpContext.Connection.Returns(connection);
+
+        _sut.Populate(_httpContext, SentryAspNetCoreOptions);
+
+        Assert.Equal(expected.ToString(), _sut.Request.Env["SERVER_PORT"]);
+    }
+
+    [Fact]
+    public void Populate_Server_SetToEnv()
+    {
+        const string expected = "kestrel.sentry";
+        var response = Substitute.For<HttpResponse>();
+        var header = new HeaderDictionary { { "Server", expected } };
+
+        response.Headers.Returns(header);
+        _httpContext.Response.Returns(response);
+
+        _sut.Populate(_httpContext, SentryAspNetCoreOptions);
+
+        Assert.Equal(expected, _sut.Request.Env["SERVER_SOFTWARE"]);
+    }
+
+    [Fact]
+    public void Populate_MachineName_SetToEnv()
+    {
+        _sut.Populate(_httpContext, SentryAspNetCoreOptions);
+
+        Assert.Equal(Environment.MachineName, _sut.Request.Env["SERVER_NAME"]);
+    }
+
+    [Fact]
+    public void Populate_NoAspNetCoreOptions_NoPayloadExtractors_NoBodyRead()
+    {
+        _sut.Populate(_httpContext, null);
+
+        _httpContext.RequestServices
+            .DidNotReceive()
+            .GetService(typeof(IEnumerable<IRequestPayloadExtractor>));
+    }
+
+    [Fact]
+    public void Populate_AspNetCoreOptionsSetTrue_NoPayloadExtractors_NoBodyRead()
+    {
+        _sut.Populate(_httpContext, SentryAspNetCoreOptions);
+
+        _httpContext.RequestServices
+            .Received(1)
+            .GetService(typeof(IEnumerable<IRequestPayloadExtractor>));
+    }
+
+    [Fact]
+    public void Populate_AspNetCoreOptionsSetTrue_PayloadExtractors_NoBodyRead()
+    {
+        var extractor = Substitute.For<IRequestPayloadExtractor>();
+        _httpContext.RequestServices
+            .GetService(typeof(IEnumerable<IRequestPayloadExtractor>))
+            .Returns(new[] { extractor });
+
+        _sut.Populate(_httpContext, SentryAspNetCoreOptions);
+
+        extractor.Received(1).ExtractPayload(Arg.Any<IHttpRequest>());
+    }
+
+    [Theory]
+    [MemberData(nameof(InvalidRequestBodies))]
+    public void Populate_PayloadExtractors_DoesNotConsiderInvalidResponse(object expected)
```

Comment:
Note, this was the only test modified.  The others were just moved to this file and cleaned up a bit.